src/goto-programs/remove_returns.cpp

```diff
@@ -163,22 +188,46 @@ bool remove_returnst::do_function_calls(
       continue;
 
     code_function_callt function_call = i_it->get_function_call();
+
     INVARIANT(
-      function_call.function().id() == ID_symbol,
+      function_call.function().id() == ID_symbol ||
+        function_call.function().id() == ID_dereference,
       "indirect function calls should have been removed prior to running "
```

Comment:
The reason for the invariant doesn't seem to be correct any more.

Reply:
Modified the comment.